rero_ils/modules/loans/cli.py

```diff
@@ -38,10 +39,166 @@
+
+
+def check_missing_fields(transaction, transaction_type):
+    """Return list of missing fields for a given transaction type.
+
+    transaction: the json transaction record.
+    transaction_type: type of transaction.
+    """
+    if transaction_type == 'checkout':
+        fields = ['item_pid', 'patron_pid', 'end_date', 'transaction_date',
+                  'transaction_location_pid', 'transaction_user_pid',
+                  'organisation', 'start_date']
+    elif transaction_type == 'request':
+        fields = ['item_pid', 'patron_pid', 'organisation', 'transaction_date',
+                  'pickup_location_pid', 'transaction_location_pid',
+                  'transaction_user_pid', 'request_expire_date']
+    elif transaction_type == 'fine':
+        fields = ['note', 'type', 'patron', 'status', 'organisation',
+                  'total_amount', 'creation_date']
+
+    return [field for field in fields if field not in transaction]
+
+
+def build_loan_record(transaction, transaction_type, item):
+    """Build the loan record before inserting into db.
+
+    transaction: the json transaction record.
+    transaction_type: type of transaction.
+    item: the item record.
+    """
+    if transaction_type == 'checkout':
+        transaction['state'] = 'ITEM_ON_LOAN'
+        transaction['trigger'] = 'checkout'
+        transaction['document_pid'] = item.document_pid
+        transaction['to_anonymize'] = False
+        transaction['item_pid'] = {'value': transaction.get('item_pid'),
+                                   'type': 'item'}
+    elif transaction_type == 'request':
+        transaction['state'] = 'PENDING'
+        transaction['trigger'] = 'request'
+        transaction['item_pid'] = {'value': transaction.get('item_pid'),
+                                   'type': 'item'}
+        transaction['document_pid'] = item.document_pid
+        transaction['to_anonymize'] = False
+
+
+@click.command('load_virtua_transactions')
+@click.option('-l', '--lazy', 'lazy', is_flag=True, default=False)
+@click.option('-e', '--save_errors', 'save_errors', type=click.File('w'))
+@click.option('-t', '--transaction_type', 'transaction_type', is_flag=False,
+              default='checkout')
+@click.option('-v', '--verbose', 'verbose', is_flag=True, default=False)
+@click.option('-d', '--debug', 'debug', is_flag=True, default=False)
+@click.argument('infile', type=click.File('r'))
+@with_appcontext
+def load_virtua_transactions(
+        infile, lazy, save_errors, transaction_type, verbose, debug):
+    """Load Virtua circulation transactions.
+
+    infile: Json Virtua transactions file.
+    transaction_type: Transaction type either checkout, request or fine.
+    :param lazy: lazy reads file.
+    :param save_errors: save error records to file.
+    """
+    # TODO: move this method and other similar methods to a new project that
+    # deals with loading data from other ILS into REROILS.
+    if save_errors:
```

Comment:
It is better to log the error and save the problematic record to a json file for later treatment as it is done for the loading of the other fixtures.